src/main/kotlin/org/jitsi/jibri/sink/impl/FileSink.kt

```diff
@@ -18,20 +18,29 @@
 package org.jitsi.jibri.sink.impl
 
 import org.jitsi.jibri.sink.Sink
+import java.lang.Integer.max
 import java.nio.file.Path
 import java.time.LocalDateTime
 import java.time.format.DateTimeFormatter
 
 /**
  * [FileSink] represents a sink which will write to a media file on the
- * filesystem
+ * filesystem.  A maximum length of 125 characters is enforced for the filename.
+ * NOTE: I considered letting the maximum length be configurable, but we require that we
+ * are able to append a timestamp to the filename, so we can't give the caller full control
+ * over the value anyway.  Because of that I just made the value hard-coded.
  */
 class FileSink(recordingsDirectory: Path, callName: String, extension: String = "mp4") : Sink {
     val file: Path
     init {
-        val currentTime = LocalDateTime.now()
-        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss")
-        val filename = "${callName}_${currentTime.format(formatter)}.$extension"
+        val timestampStr = LocalDateTime.now().format(TIMESTAMP_FORMATTER)
+        // How much of the call name we'll use is determined by how many characters are leftover
+        // counting the timestamp and underscore separator
+        val maxCallNameLength = MAX_FILENAME_LENGTH - timestampStr.length - 1
```

Comment:
Is this -1 to account for the "_"? I don't know whether it's important to us exactly 125 charactrs, or just any limit. I don't understand why the extension is considered special.

I would suggest counting the extension, and doing something like
```
val postfix = "_${currentTime.format(formatter)}.$extension"
val filename = "${callName.take(MAX_FILENAME_LENGTH - postfix.length)}${postfix}"
```

Reply:
I tried many iterations of this entire thing.  The `-1` is indeed to cover the `_`.  125 itself is arbitrary.  I wanted this to at least roughly look like an implementation of enforcing a true filesize limit (rather than an arbitrary size) and I googled around and _think_ the extension doesn't count against that.  

But, I like your suggestion so will use that.